akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -51,12 +51,15 @@ private[http] object HttpServerBluePrint {
     stack(settings, remoteAddress, log)
 
   def stack(settings: ServerSettings, remoteAddress: Option[InetSocketAddress], log: LoggingAdapter)(implicit mat: Materializer): Http.ServerLayer = {
-    userHandlerGuard(settings.pipeliningLimit) atop
-      requestPreparation(settings, remoteAddress) atop
-      controller(settings, log) atop
-      parsingRendering(settings, log) atop
-      websocketSupport(settings, log) atop
-      unwrapTls
+    val theStack =
+      userHandlerGuard(settings.pipeliningLimit) atop
+        requestPreparation(settings) atop
```

Comment:
I agree, but I'd rather choose the lesser of 2 evils 😊

-- 
Cheers,
√
On 2 Dec 2015 13:05, "Konrad Malawski" <notifications@github.com> wrote:

> In
> akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala
> <https://github.com/akka/akka/pull/19060#discussion_r46406253>:
>
> > @@ -51,12 +51,15 @@ private[http] object HttpServerBluePrint {
> >      stack(settings, remoteAddress, log)
> >
> >    def stack(settings: ServerSettings, remoteAddress: Option[InetSocketAddress], log: LoggingAdapter)(implicit mat: Materializer): Http.ServerLayer = {
> > -    userHandlerGuard(settings.pipeliningLimit) atop
> > -      requestPreparation(settings, remoteAddress) atop
> > -      controller(settings, log) atop
> > -      parsingRendering(settings, log) atop
> > -      websocketSupport(settings, log) atop
> > -      unwrapTls
> > +    val theStack =
> > +      userHandlerGuard(settings.pipeliningLimit) atop
> > +        requestPreparation(settings) atop
>
> Well ok, I can give the Source approach a shot so we can compare the two,
> but still feel it's wrong.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/19060/files#r46406253>.
>